### DISCUSSION TITLE
Minimal init.lua for bug report reproduction steps

### DISCUSSION THREAD

**Discussion by mrjones2014** [mrjones2014/legendary.nvim#93] [Announcements]
For reporting bugs, you can base your minimal `init.lua` on the following:

```lua
local root = vim.fn.fnamemodify('./.repro', ':p')

-- set stdpaths to use .repro
for _, name in ipairs({ 'config', 'data', 'state', 'cache' }) do
  vim.env[('XDG_%s_HOME'):format(name:upper())] = root .. '/' .. name
end

-- bootstrap lazy
local lazypath = root .. '/plugins/lazy.nvim'
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    'git',
    'clone',
    '--filter=blob:none',
    '--single-branch',
    'https://github.com/folke/lazy.nvim.git',
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  -- do not remove the colorscheme! it makes testing nicer
  'folke/tokyonight.nvim',
  {
    'mrjones2014/legendary.nvim',
    dependencies = {
      -- place your `vim.ui.select()` handler plugin here
      -- for example:
      -- 'nvim-telescope/telescope.nvim',
      -- 'stevearc/dressing.nvim',
    },
  },
  -- add any other pugins here
}

require('lazy').setup(plugins, {
  root = root .. '/plugins',
})

require('dressing').setup()

require('legendary').setup({
  -- add any options here
})

-- add anything else here
vim.opt.termguicolors = true
-- do not remove the colorscheme! it makes testing nicer
vim.cmd([[colorscheme tokyonight]])
```

**Comment by tsyroid** [comment]
This is very cool, "non-jank" stuff MrJones. I've used some similar "minimal init" files before, but they were more generic. This not only specifically targets Legendary, it also clearly shows me HOW to customize it for Legendary -- which cycles back to your blog post about writing code without requiring comments, and my needs a few weeks back to understand better the bootstrap process.

Short story: Something like this needs to be bundled within every plugin's tree somewhere.

Aside: I clearly need to find a better way to track/follow what's going on in repos I'm interested in. I only discovered this today, 8 days after it was posted.

**Reply by mrjones2014** [reply]
This isn’t in the repo tree because it’s only really needed when trying to reproduce issues that are hard to reproduce. Didn’t seem necessary to have in the actual repo, thus I put it here in discussions. I link to this from the new issue template (you’ll see if you click “New Issue” then select “Bug Report”).

And credit where credit is due, I pretty much took this from Telescope.nvim’s issue template and customized it to be for legendary instead of telescope.

**Comment by tsyroid** [comment]
All good. Just noting or commenting more on the wonder of Github. So much
here. So much to be discovered. But we’re all still trying to find our way
on how to do that in a way that reflects our personal levels of want/need.
Struggle with this issue every day.

It just becomes important to call-out when I see something that fills a
gap, that I didn’t know I had, on a thread a missing, that was buried
somewhere in this byzantine called GitHub.

Best,
/tom

**Comment by danielo515** [comment]
Can I copy this to any random file on my disk and start nvim with `nvim -u path/to/this/init.lua` and get it working without affecting my main nvim install?

**Reply by mrjones2014** [reply]
You'll wanna do `nvim --clean -u path/to/this/init.lua`, the `--clean` part will avoid Neovim loading your plugins from your config

**Reply by danielo515** [reply]
Thanks for your help. I'm trying to follow your steps,and I am getting this error:

```
Error executing vim.schedule lua callback: ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:87: Vim(call):E482: Can't open file /path/to/manifest f
or writing: no such file or directory
stack traceback:
        [C]: in function 'cmd'
        ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:87: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
which-key.nvim not available. If you are lazy-loading, be sure that which-key.nvim is added to runtime path before running legendary.nvim config.
Error executing vim.schedule lua callback: /tmp/nvim/site/pack/packer/start/packer.nvim/lua/packer.lua:359: Vim(lua):Error creating utility commands with lege
ndary: /Users/danielo/testing/legendary-reproduce.lua:40: attempt to call local 'bind_command' (a nil value)
stack traceback:
        [C]: in function 'cmd'
        /tmp/nvim/site/pack/packer/start/packer.nvim/lua/packer.lua:359: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

Do you know who is referring to that path to manifest? Is some config I forgot to ad?

**Reply by danielo515** [reply]
I see that, for some reason, packer fails to install/update legendary
![image](https://user-images.githubusercontent.com/2270425/202374383-7ebc20ef-a4e7-4203-b925-b9668f79ea48.png)
I have it setup using the specific commit I use in my normal configuration, so I do not know what may be wrong

**Reply by mrjones2014** [reply]
> Can't open file /path/to/manifest

where’s this path coming from? It seems like there’s a placeholder path somewhere in the config.

**Reply by danielo515** [reply]
I don't know where that placeholder is coming from. I grepped the entire install folder and nothing appears there. I shared my config file in a gist in the issue I opened. Maybe you can try it yourself and see if you get the same error.

**Reply by mrjones2014** [reply]
I figured it out, it’s a in the minimal init. I’ve edited it. Should work now.

**Reply by danielo515** [reply]
May I see a diff of the changes? I would love to patch it in my local copy without having to start from scratch.
Thanks!

**Reply by danielo515** [reply]
Ok, maybe is the addition of `vim.g.loaded_remote_plugins = ""` ?
How is that affecting in such a drastic way?

**Reply by mrjones2014** [reply]
It has something to do with the way that Neovim loads remote plugins, I think. Not sure exactly.

**Reply by mrjones2014** [reply]
Yes that is the line I changed.

**Comment by towry** [comment]
How to use this minimal file?

**Reply by towry** [reply]
1. save to `/tmp/init.lua`
2. `nvim -u /tmp/init.lua`
